### DISCUSSION TITLE
Is there an easy way to create multiple PRs at once?

### DISCUSSION THREAD

**Discussion by bradneuman** [VirtusLab/git-machete#1362] [Q&A] [answered]
I'm coming from tools like `git spr` and a common workflow for me on a new feature is to make a feature branch and then interactively rebase it into multiple PRs (to keep each PR small and related). Then I'll push these all up at once, and may continue working on this chain / stack as I wait for feedback.

I'm looking for something like `git machete github create-pr --all` which, rather than creating a single PR containing all branches machete knows about, would create one PR per branch. E.g., if I have this:

```
  main
  │
  └─brad/feat1
    │
    └─brad/sub-feat-on-1
      │
      └─brad/feat-2
```

and each of these branches is not tracking an upstream, nor pushed, nor has a PR, then I run `git machete github create-pr` what I get is:
```
  main
  │
  └─brad/feat1
    │
    └─brad/sub-feat-on-1
      │
      └─brad/feat-2  PR #523
 ```

Where 523 is a single PR that contains all commits from all 3 of those branches. What I want is something like this: 
```
  main
  │
  └─brad/feat1  PR #523
    │
    └─brad/sub-feat-on-1  PR #524
      │
      └─brad/feat-2  PR #525
 ```
With separate stacked PRs for each named branch.

I can do this manually via `git machete go` and `create-pr`s one at a time, but that's a bit annoying and possibly error prone if I miss one.

Alternatively, I think this suggestion to support this behavior during `traverse` might work for me, if it were implemented: https://github.com/VirtusLab/git-machete/issues/1241

**Comment by ccoVeille** [comment]
I have no idea if git-machete could support this

But thanks for letting me know git spr

I assume it's this project
https://github.com/ejoffe/spr

I will give it a try

**Reply by bradneuman** [reply]
Yeah that's the one. It has some pros but I'm ultimately looking to move away from it because:
* 1 commit == 1 PR makes it hard for reviewers to follow what's going on (you need to force push)
* It seems to leave tons of dead branches around, both locally and on github, which is a bit annoying
* It isn't super duper stable and can get into weird states that can be a little tricky to recover from (mostly I think this was because I reordered some commits)

But its github integration is really nice

**Comment by PawelLipski** [comment, marked as answer]
Hey, this will be implemented soon as a part of `traverse` flow, controlled by a flag (something like `--sync-github-prs`)

**Reply by bradneuman** [reply]
Thanks, looking forward to trying it

**Comment by PawelLipski** [comment]
Update, there's already `git machete traverse --sync-github-prs` that allows for retargeting PRs from `traverse` (PR #1340, to be released as v3.32.0 soon). #1241 will follow soon

**Comment by PawelLipski** [comment]
Long done at this point
